base/error.jl

```diff
@@ -49,3 +49,34 @@ macro assert(ex, msgs...)
     end
     :($(esc(ex)) ? $(nothing) : throw(Main.Base.AssertionError($msg)))
 end
+
+
+macro throw_argument_error(msg)
```

Comment:
That makes a lot of sense.
Done: https://github.com/samoconnor/julia/commit/258aa7458f174402f4b689de450f24b645dc64ea